### ISSUE TITLE
Is there an rank requirement option for a certain money amount?

### ISSUE THREAD

**Issue by stifflered** [Zrips/CMI#585]
I'm not seeing it in the documentation or anything. I'm looking for a way to auto-rankup a player when they reach a certain amount of Vault money.

**Comment by RustyDagger** [comment]
Iv not really used the Rank system but Id imagine the closest you might get to that is setting a price to Rankup then running a command to give them that money back once they get the rank. then maybe on a few of the events in the events file from cmi try force them to rankup? Im not sure how your system works so that may not be a viable option just an idea.

**Comment by Guddina-tim** [comment]
Yeah, that doesn't quite work because I still need to have some way to check the money value that they have before it triggers the rankup. Thanks for trying, though.

**Comment by RustyDagger** [comment]
Why? Set the cost to get the rank to 90k then when they get 90k it would give them the rank, then run the command to give the 90k back to them. 

The price of the rank does the check for you.

**Comment by Guddina-tim** [comment]
Wait, so the cost will act on its own as a check? I didn't know that. I had assumed the cost would be taken from then when they hit the rank through some other means. I'll give that a shot!

**Comment by MCTyler** [comment]
No need for all that @RustyDagger Ranks can be configured for a cost that is pulled directly from CMI Economy...I.E.
```
Coal:
  Enabled: true
  DisplayName: "&0Coal"
  AutoRankup: true
  RankupConfirmation: false
  NextRanks:
  - Iron
  MoneyCost: 25
```
Then run the command to give the money back if that's what you desire.

**Comment by Guddina-tim** [comment]
I have no gotten this to work. I don't just want to check if they have a certain amount of money AND other requirements. I want the ONLY requirement to be "Have they made $X total in the history of their account?"

It's also important that it's a total amount, not just a "current" amount. The reason: I have a feature that let's early game players make a little extra money (up to $5,000) to get them started. But, this feature I've added is super easy to automate and farm a lot of money if left unchecked. Let's say that this rankup runs only based on the current amount a player has, then they could automatically farm $4,999, stop the farm, buy a bunch of diamond ore and get their balance back down to $0, and then farm up to $4,999 again. Rinse and repeat.

I tried using the method described by both of you, and it did not work. I need other requirements included besides money in order to trigger the rankup, and I can't base this on the lifetime-gained money amount. A placeholder statistic that tracks the lifetime-gained currency, and a stats requirement based on that, would be perfect.

But, it looks like CMI doesn't have this. I'll figure out something else.
